tests/integration/states/test_network.py

```diff
@@ -25,18 +30,130 @@ def setUp(self):
             )
+        else:
+            if os_family == 'Debian':
+                self.run_function('cmd.run', cmd='cp -p /etc/network/interfaces /etc/network/interfaces.bkp')
+
+    def tearDown(self):
+        # Remove "iface_name" test interface
+        os_family = self.run_function("grains.get", ["os_family"])
+        self.run_function('ip.down', iface=iface_name, iface_type='eth')
+        if os_family == 'Debian':
+            self.run_function('cmd.run', cmd='mv /etc/network/interfaces.bkp /etc/network/interfaces')
+        else:
+            self.run_function('cmd.run', cmd='rm /etc/sysconfig/network-scripts/ifcfg-{}'.format(iface_name))
+
+    def addInterface(self, testFlag):
+        """
+        Add network interface. Used on tests when needed.
+        """
+        ret = self.run_state('network.managed',
+                          name=iface_name,
+                          enabled=iface_enabled,
+                          proto=iface_proto,
+                          ipaddr=iface_ipaddr,
+                          test=testFlag
+                      )
+        
+        return ret
 
     @pytest.mark.slow_test
-    def test_managed(self):
+    def test_managed_addInterface0(self):
         """
-        network.managed
+        network.managed add new interface with test flag
         """
-        state_key = "network_|-dummy0_|-dummy0_|-managed"
+        # Execute test
+        ret = self.addInterface(True)
 
-        ret = self.run_function("state.sls", mods="network.managed", test=True)
-        self.assertEqual(
-            "Interface dummy0 is set to be added.", ret[state_key]["comment"]
-        )
+        # and validate results
+        self.assertSaltNoneReturn(ret)
+        self.assertInSaltComment('Interface {} is set to be added.'.format(iface_name), ret)
+
+    @pytest.mark.slow_test
+    def test_managed_addInterface1(self):
+        """
+        network.managed add new interface
+        """
+        # Execute test
+        ret = self.addInterface(False)
+
+        # and validate results
+        self.assertSaltTrueReturn(ret)
+        self.assertInSaltComment('Interface {} added.'.format(iface_name), ret)
+
+    @pytest.mark.slow_test
+    def test_managed_existingInterfaceNoChanges0(self):
+        """
+        network.managed with existing interfaces without changing configs with test flag
+        """
+        # Add test interface
+        self.addInterface(False)
+
+        # Execute test
+        ret = self.run_state('network.managed',
+                                name=iface_name,
+                                enabled=iface_enabled,
+                                proto=iface_proto,
+                                ipaddr=iface_ipaddr,
+                                test=True
+                            )
+
+        # and validate results
+        self.assertSaltTrueReturn(ret)
+        self.assertInSaltComment('Interface {} is up to date.'.format(iface_name), ret)
```

Comment:
It looks like these tests rely on the previous tests running before them. Is that assumption correct? Is so we want to ensure that is not the case.

Reply:
I did not find a lot of documentation on how to setup the tests or use stuff (e.g. salt_master, salt_call_cli etc) with the new pytests but figured out most from pytest docs and the current files inside the `tests/pytests` folder. I moved the tests there and results are still good.

Related to the concern about the dependency of tests, they do not rely on previous. Instead, I put in place a common function to add the same stuff on each test when needed and used the fixture for the things that are common across all of them.

```
==================================================================================== test session starts ====================================================================================
platform linux -- Python 3.9.2, pytest-7.2.0, pluggy-0.13.0 -- /repo/.nox/test-parametrized-3-crypto-none-transport-zeromq-coverage-false/bin/python
cachedir: .pytest_cache
tempdir: /tmp/stsuite
max open files; soft: 3072; hard: 1048576
rootdir: /repo, configfile: pytest.ini
plugins: subtests-0.4.0, skip-markers-1.2.0, custom-exit-code-0.3.0, helpers-namespace-2021.4.29, shell-utilities-1.6.0, system-statistics-1.0.2, timeout-1.4.2, flaky-3.7.0, httpserver-1.0.4, tempdir-2019.10.12, salt-factories-1.0.0rc21
collected 10 items                                                                                                                                                                          

tests/pytests/integration/states/test_network.py::TestNetwork::test_managed_addInterface0 PASSED
tests/pytests/integration/states/test_network.py::TestNetwork::test_managed_addInterface1 PASSED
tests/pytests/integration/states/test_network.py::TestNetwork::test_managed_existingInterfaceNoChanges0 PASSED
tests/pytests/integration/states/test_network.py::TestNetwork::test_managed_existingInterfaceNoChanges1 PASSED
tests/pytests/integration/states/test_network.py::TestNetwork::test_managed_existingInterfaceWithChanges0 PASSED
tests/pytests/integration/states/test_network.py::TestNetwork::test_managed_existingInterfaceWithChanges1 PASSED
tests/pytests/integration/states/test_network.py::TestNetwork::test_managed_existingInterfaceWithChanges2 PASSED
tests/pytests/integration/states/test_network.py::TestNetwork::test_managed_existingInterfaceWithChanges3 PASSED
tests/pytests/integration/states/test_network.py::TestNetwork::test_routes PASSED
tests/pytests/integration/states/test_network.py::TestNetwork::test_system PASSED

===================================================================================== warnings summary ======================================================================================
tests/pytests/integration/states/test_network.py::TestNetwork::test_managed_addInterface0
  /repo/.nox/test-parametrized-3-crypto-none-transport-zeromq-coverage-false/lib/python3.9/site-packages/napalm/base/helpers.py:8: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3, and in 3.10 it will stop working
    from collections import Iterable

tests/pytests/integration/states/test_network.py::TestNetwork::test_managed_addInterface0
  /repo/.nox/test-parametrized-3-crypto-none-transport-zeromq-coverage-false/lib/python3.9/site-packages/jnpr/junos/factcache.py:10: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3, and in 3.10 it will stop working
    class _FactCache(collections.MutableMapping):

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
======================================================================== 10 passed, 2 warnings in 168.34s (0:02:48) =========================================================================
nox > Session test-parametrized-3(crypto=None, transport='zeromq', coverage=False) was successful.
nox > Ran multiple sessions:
nox > * test-3(coverage=False): success
nox > * test-parametrized-3(crypto=None, transport='zeromq', coverage=False): success
```